Content.Server/SimpleStation14/Magic/Asclepius/Systems/AsclepiusSystem.Healing.cs

```diff
@@ -0,0 +1,72 @@
+using Robust.Shared.Utility;
+
+namespace Content.Server.SimpleStation14.Magic.Asclepius.Systems
+{
+    public sealed class AsclepiusHealingSystem : EntitySystem
+    {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+        [Dependency] private readonly EntityLookupSystem _lookupSystem = default!;
+        [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var component in _entityManager.EntityQuery<AsclepiusStaffComponent>()
+            .Where(x =>
+                x.BoundTo != EntityUid.Invalid &&
+                _entityManager.TryGetComponent<HippocraticOathComponent>(x.BoundTo, out var _)
+            ))
+            {
+                foreach (var entity in _lookupSystem.GetEntitiesInRange(component.Owner, 7f, LookupFlags.Uncontained)
+                .Where(y =>
+                    _entityManager.TryGetComponent<DamageableComponent>(y, out var damageable) &&
+                    damageable.TotalDamage > 0f
+                ))
+                {
+                    var oath = _entityManager.GetComponent<HippocraticOathComponent>(component.BoundTo);
+
+                    oath.HealingAccumulator += frameTime;
+                    if (oath.HealingAccumulator < oath.HealingTime)
+                    {
+                        continue;
+                    }
+                    oath.HealingAccumulator = 0f;
+
+                    foreach (var (type, amount) in oath.DamageTypes)
+                    {
+                        if (!_prototypeManager.TryIndex<DamageGroupPrototype>(type, out var group))
+                        {
+                            if (!_prototypeManager.TryIndex<DamageTypePrototype>(type, out var proto))
+                            {
+                                DebugTools.Assert($"[Asclepius Healing]: Damage type {type} does not exist!");
+                                continue;
+                            }
+
+                            _damageableSystem.TryChangeDamage(component.BoundTo, new DamageSpecifier(proto, -amount));
+                        }
+                        else
+                        {
+                            foreach (var damageType in group.DamageTypes)
+                            {
+                                if (!_prototypeManager.TryIndex<DamageTypePrototype>(damageType, out var proto))
+                                {
+                                    DebugTools.Assert($"[Asclepius Healing]: Damage type {damageType} does not exist!");
+                                    continue;
+                                }
+
+                                _damageableSystem.TryChangeDamage(component.BoundTo, new DamageSpecifier(proto, -(amount / group.DamageTypes.Count)));
```

Comment:
This is very interesting, but it's probably worth limiting it to some extent. Healing more major / mystical damage types would be good, like clearing up bleeding very quickly and healing asphyx to make them an invaluable role in an ER, or cleansing toxin. Not being as effective at something like burns could be thematic with the nature of the staff, but is just a thought.